aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2051,7 +2051,7 @@ impl AptosVM {
         for write in module_write_set.writes().values() {
             // It is sufficient to simply get the size in order to enforce read-before-write.
             module_storage
-                .fetch_module_size_in_bytes(write.module_address(), write.module_name())
+                .unmetered_get_module_size(write.module_address(), write.module_name())
```

Comment:
hm, processing in module_write_set assumes module read is already metered due to read-before-write, but then here we make an unmetered read to satisfy read-before-write. How is it possible to actually have a module write without already having read the path? it would be needed for upgradability checks, etc, right?

I suppose this is for write-set transactions? But then it'd be great if the code flow somehow enforced / demonstrated that... Or we could at least add a comment to read-write-set or here, and also explain that metering isn't done in this case?

Reply:
This is for writeset txns which are unmetered, adding a comment here for that. `module_write_set` is passed with a txn payload, so nothing has been read yet.